schema-registry-mock/src/main/java/com/bakdata/schemaregistrymock/SchemaRegistryMock.java

```diff
@@ -127,25 +127,31 @@ public SchemaRegistryMock() {
         this(null);
     }
 
-    private static UrlPattern getSchemaPattern(final Integer id) {
-        return WireMock.urlPathEqualTo(SCHEMA_BY_ID_PATTERN + id);
+    private static MappingBuilder getSchema(final int id) {
+        return WireMock.get(WireMock.urlPathEqualTo(SCHEMA_BY_ID_PATTERN + (Integer) id))
+                .withQueryParam("fetchMaxId", WireMock.matching(BOOLEAN_PATTERN));
     }
 
-    private static UrlPattern getDeleteSubjectPattern(final String subject) {
-        return WireMock.urlEqualTo(ALL_SUBJECT_PATTERN + "/" + subject + "?permanent=false");
+    private static MappingBuilder deleteSubject(final String subject) {
+        return WireMock.delete(WireMock.urlPathEqualTo(ALL_SUBJECT_PATTERN + "/" + subject))
+                .withQueryParam("permanent", WireMock.equalTo("false"));
     }
 
-    private static UrlPattern getSubjectSchemaVersionPattern(final String subject) {
-        return WireMock.urlPathMatching(ALL_SUBJECT_PATTERN + "/" + subject
-                + "(?:\\?(?:deleted|normalize)=(?:true|false)(?:&(?:deleted|normalize)=(?:true|false))*)?");
+    private static MappingBuilder postSubjectVersion(final String subject) {
+        return WireMock.post(WireMock.urlPathMatching(ALL_SUBJECT_PATTERN + "/" + subject))
+                .withQueryParam("deleted", WireMock.matching(BOOLEAN_PATTERN))
+                .withQueryParam("normalize", WireMock.matching(BOOLEAN_PATTERN));
     }
 
-    private static UrlPattern getSubjectVersionsPattern(final String subject) {
-        return WireMock.urlEqualTo(ALL_SUBJECT_PATTERN + "/" + subject + "/versions");
+    private static MappingBuilder getSubjectVersions(final String subject) {
+        return WireMock.get(WireMock.urlPathEqualTo(ALL_SUBJECT_PATTERN + "/" + subject + "/versions"))
+                .withQueryParam("deletedOnly", WireMock.matching(BOOLEAN_PATTERN))
+                .withQueryParam("deleted", WireMock.matching(BOOLEAN_PATTERN));
     }
 
-    private static UrlPathPattern getSubjectVersionPattern(final String subject) {
-        return WireMock.urlPathMatching(ALL_SUBJECT_PATTERN + "/" + subject + "/versions/(?:latest|\\d+)");
+    private static MappingBuilder getSubject(final String subject) {
+        return WireMock.get(
+                WireMock.urlPathMatching(ALL_SUBJECT_PATTERN + "/" + subject + "/versions/(?:latest|\\d+)"));
```

Comment:
Why do you use `urlPathMatching`, and `urlPathEqualTo` with `withQueryParam` above?

Reply:
Because here we have a regex (`(?:latest|\\d+)`) while above not. Also here there are no query params